### ISSUE TITLE
zstd.h not found when building on Mac

### ISSUE THREAD

**Issue by jaffee** [citusdata/citus#7196]
I'm following the instructions in CONTRIBUTING.md for building on a mac. `./configure` runs fine, but then `make` gives me `columnar_compression.c:32:10: fatal error: 'zstd.h' file not found`

I have `zstd.h` in `/usr/local/include`, and indeed if I go into `src/backend/columnar` and run the command which make was trying to run (see below), but with the addition of `-I/usr/local/include`, then it runs successfully.

```
clang -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2  -std=gnu99 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Wno-gnu-variable-sized-type-not-at-end -Wno-declaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wmissing-declarations -Wmissing-prototypes -Wshadow -Werror=vla -Werror=implicit-int -Werror=implicit-function-declaration -Werror=return-type -DGIT_VERSION=\""main(sha: e0683aab8)"\" -I '/Users/jaffee/citus/src/include' -I'../../../src/include' -I/usr/local/include/postgresql@14 -I/Users/jaffee/citus/vendor/safestringlib/include -I. -I./ -I/usr/local/include/postgresql@14/server -I/usr/local/include/postgresql@14/internal -I/usr/local/Cellar/icu4c/73.2/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/usr/local/opt/openssl@3/include -I/usr/local/opt/readline/include -I/usr/local/Cellar/lz4/1.9.4/include   -c -o columnar_compression.o columnar_compression.c
```

So it seems like zstd is being detected by `configure`, but not appropriately accounted for by `make`. Apologies if this is a trivial issue... it's been a long time since I've had to find my way around a C build system.

System Details:
Monterey 12.4 2019 Intel Macbook Pro

Dependency versions (from brew):
==> postgresql@14: stable 14.9 (bottled)
==> python@3.11: stable 3.11.5 (bottled)
==> zstd: stable 1.5.5 (bottled), HEAD
==> git: stable 2.42.0 (bottled), HEAD

TIA for any assistance!

**Comment by jaffee** [comment]
btw, I did get it to work with `./configure --without-zstd`

**Comment by thanodnl** [comment]
Sorry for the late reply, issue didnt' bubble up to me until now.

I used to work on a mac, and also installed all dependencies via brew. I don't exactly remember how I found it, I believe it was brew prompting me to do so, but to make the compilation work I have added many `CLFAGS`/`CPPFLAGS`/`LDFLAGS`/`PKG_CONFIG_PATH` to my `.profile` over time.

The ones that might be of importance for you are:

```
export LDFLAGS="-L/usr/local/opt/lz4/lib $LDFLAGS"
export CPPFLAGS="-I/usr/local/opt/lz4/include $CPPFLAGS"
export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/local/opt/lz4/lib/pkgconfig"

export LDFLAGS="-L/usr/local/opt/zstd/lib $LDFLAGS"
export CPPFLAGS="-I/usr/local/opt/zstd/include $CPPFLAGS"
export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/local/opt/zstd/lib/pkgconfig"

export CFLAGS="-I/usr/local/include/ $CFLAGS"
export CPPFLAGS="-I/usr/local/include $CPPFLAGS"
export PATH="/usr/local/sbin:$PATH"
```

Others I have are for `icu4u`, `openssl`, `readline` and `gettext`. This environment has grown over the course of 5 years and I don't know why each of these were added, but compilation with them should work. At least it does for me.

I have since moved away from development directly on osx in favour of devcontainers. If that works for you its contains a curated environment with all tools we use. It's a quick and consistent way to setup the environment to develop, debug and run our test suite. If you are interested a description on how to use it is in [CONTRIBUTING.md](https://github.com/citusdata/citus/blob/main/CONTRIBUTING.md#devcontainer--github-codespaces).
